Feature/single project

csturm83 opened the pull request:
**Summary**
Target/build/package multiple frameworks from a single .csproj/.sln.

This PR fixes/implements the following **features**:

* [ ] Single primary .csproj
* [ ] Build multiple targets at once
* [ ] Use netstandard 1.1 for PCL target

What existing problem does the pull request solve?
 - Targeting multiple frameworks.
 - De-cluttering root folder a bit

**Test plan**
No functional changes. All existing tests pass.

**Closing issues**
Closes #188
Closes #190

**_Note:_** There is likely some followup cleanup of .bat and .nuspec files needed if this PR is accepted.

csturm83 commented on QRCoder/Base64QRCode.cs:
```diff
@@ -1,4 +1,5 @@
-﻿using System;
+﻿#if !NETSTANDARD1_1
```
Thinking about this a bit more, I think this might affect NET45 targets (which is included in NS1.1). May need to add a NET45 target explicitly to the csproj to avoid this. Will take a look once I have access to my dev machine.

csturm83 commented on QRCoder/QRCodeGenerator.cs:
```diff
@@ -402,7 +402,7 @@ public static int MaskCode(ref QRCodeData qrCode, int version, ref List<Rectangl
 
 
-                #if NET40
+                #if NET35 || NET40
```
These ```NET35 || NET40``` directives should probably just be changed to ```NETFRAMEWORK```.

csturm83 commented on QRCoder/QRCoder.csproj:
```diff
@@ -33,4 +35,8 @@
     <PackageReference Include="System.Drawing.Common" Version="4.5.1" />
   </ItemGroup>
 
+  <PropertyGroup>
```
Workaround for ```dotnet pack``` multi-targeting net35. https://github.com/Microsoft/msbuild/issues/1333#issuecomment-296346352

csturm83 commented:
I was able to successfully create a .nupkg using the updated .csproj and MyGet.bat. 

Was getting an obscure ```error MSB4057: The target "pack" does not exist in the project.```, but it worked.

Note that you can also pack the .csproj directly from Visual Studio.

jnyrup commented:
A net45 project will use net40 - [link](http://nugettoolsdev.azurewebsites.net/4.8.0/get-nearest-framework?project=net45&package=net35%0D%0Anet40%0D%0Anetstandard1.1%0D%0Anetstandard2.0)
A net461 project will also use net40 - [link](http://nugettoolsdev.azurewebsites.net/4.8.0/get-nearest-framework?project=net461&package=net35%0D%0Anet40%0D%0Anetstandard1.1%0D%0Anetstandard2.0)

csturm83 commented:
Thanks @jnyrup for the feedback! Removed the net45 target, since it is not necessary. Updated ```MyGet.bat``` to create the NuGet package cleanly.

I realized the ```error MSB4057: The target "pack" does not exist in the project.``` error I was getting was because I was attempting to pack the console/demo projects.

@codebude , any remaining concerns?

jnyrup commented:
Given all the fiddling with preprocessor directives, you would probably also want to ensure that the tests are run for all target frameworks.

I've made a branch that creates tests for all four target frameworks.
https://github.com/csturm83/QRCoder/compare/feature/single-project...jnyrup:feature/SharedProject?expand=1

csturm83 commented:
I've updated the QRCoderTests.csproj to multi-target (in a single test project). Added testing of all targets to MyGet.bat. The 4.5.2 target passes all tests, but other targets are failing (#193).

ps30 commented:
Hi question - Will this single project model support the current .NET 4.8?  Assume net40 will also work for .NET 4.8

csturm83 commented:
Correct. NuGet will resolve net40 for .NET 4.8. Technically, netstandard2.0 would work as well.

csturm83 commented:
@codebude, any update on this one? Ready for merge?

ps30 commented:
@codebude and @csturm83 -- With .Net Core 3.0 out now can this be also be supported with this single project?  Also looking for a new build with this PR merged.

csturm83 commented:
The netstandard2.0 target should cover .NET Core 3.0.

codebude commented:
@csturm83 @jnyrup thanks for the stunning work you've done here. And sorry for that longtime without answers. I don't want to see this project die and at the other hand I'm stuck so deep in my main job the last month, that I just couldn't keep up with this project. 

In general I really appreciate and understand what you want to achieve with this changes. Nevertheless I'm a little bit scared to merge them, because I fear it could break my build chain. (Every commit to the Git repo, will trigger a pull + build on myget.org via webhooks. The builds on myget then will be used to generate and publish the Nuget packages. From my experience in the past, often the slightest changes in the project structure or the myget.bat broke the build chain...)

Nevertheless - since the project is out of date for so long - I'll accept the challenge (because I see the good things in your changes) and will just merge it. If it breaks the build chain - I'll just work on fixing it. So far - thanks again! And now let's see how the merge comes out. ;-)

codebude commented:
It broke while building the test project for netcoreapp2.0:

```
C:\Program Files\dotnet\sdk\2.2.101\Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3245: 
Could not resolve this reference. Could not locate the assembly "NDesk.Options". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
[C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj] Program.cs(5,7): error CS0246: The type or namespace name 'NDesk' could not be found (are you missing a using directive or an assembly reference?) 
[C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj] Program.cs(231,38): error CS0246: The type or namespace name 'OptionSet' could not be found (are you missing a using directive or an assembly reference?) 
[C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj]
...
```

If you have any clue @csturm83  - let me know. Meanwhile I'll try to figure out what's going on. Having left this project aside for too long time. :-(

jnyrup commented:
@codebude can you share a more complete build log and the build script?
It seems odd why `QRCoderTests` should fail building - it is not referencing `QRCoderConsole` or `NDesk.Options`.

My first guess is that the NDesk.Options package is not restored.

codebude commented:
@jnyrup this is the full build log:

```powershell
MyGet Wonka - Version 1.9.0.17180
(C) MyGet 2012 - 2020 - www.myget.org
Making NuGet for all of the world to enjoy! (@jglozano)
    - http://twitter.com/jglozano/status/228462337663832064


Verifying package source...
Package source verification succeeded.


Initializing build environment...
Build environment variables:
  BuildRunner=MyGet
  MsBuildExe=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\MSBuild.exe
  GallioEcho=C:\Program Files\Gallio\bin\Gallio.Echo.exe
  XUnit192Path=C:\Wonka\testrunners\xunit.runners.1.9.1\tools
  XUnit20Path=C:\Wonka\testrunners\xunit.runner.console.2.0.0\tools
  VsTestConsole=C:\Wonka\testrunners\VSTest.Console.12.0.30723.0\lib\vstest.console.exe
  MsTestExe=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\..\..\..\Common7\IDE\MSTest.exe
  SourcesPath=C:\a\7ae0494
  NuGet=C:\Wonka\nuget\NuGet.CommandLine\tools\NuGet.exe
  NpmExePath=C:\users\wonka\AppData\Roaming\npm\npm.cmd
  Configuration=Release
  Targets=Rebuild
  Platform=
  VersionFormat=1.3.7-CI{0}
  BuildCounter=1
  PrereleaseTag=CI00001
  PackageVersion=1.3.7-CI00001
  EnableNuGetPackageRestore=true


Initializing git-based build environment...
Additional build environment variables:
  GitPath=C:\Program Files\Git\bin\git.exe
  GitVersion=C:\ProgramData\Chocolatey\lib\GitVersion.Portable\tools\gitversion.exe
  PATH=C:\Python27\;C:\Python27\Scripts;C:\ProgramData\Oracle\Java\javapath;C:\ruby200\bin;C:\ruby193\bin;C:\ruby192\bin;C:\ruby187\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Microsoft ASP.NET\ASP.NET Web Pages\v1.0\;C:\Program Files (x86)\nant-0.92\bin;C:\Program Files\Microsoft SQL Server\110\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\100\Tools\Binn\VSShell\Common7\IDE\;C:\Program Files (x86)\Microsoft SQL Server\100\Tools\Binn\;C:\Program Files\Microsoft SQL Server\100\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\100\DTS\Binn\;C:\Program Files\Gallio\bin;C:\Users\MyGetTeam\AppData\Roaming\npm;C:\Installs\scriptcs;C:\Program Files (x86)\git\bin;C:\Program Files\Mercurial;C:\Program Files\SlikSVN\bin;C:\Program Files\Microsoft SQL Server\100\DTS\Binn\;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\;C:\Program Files\Microsoft\Web Platform Installer\;C:\ProgramData\chocolatey\bin;C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.0\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files (x86)\nodejs;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Microsoft Emulator Manager\1.0\;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Go\bin;C:\Program Files\dotnet\;C:\Program Files\Git\cmd;C:\Program Files\Mercurial\;C:\Users\Wonka\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update2\bin;C:\Users\Wonka\.dnx\bin;C:\Users\Wonka\.k\runtimes\kre-clr-win-x86.1.0.0-beta4-11203\bin;C:\Users\Wonka\.dotnet\runtimes\dotnet-clr-win-x86.1.0.0-beta4-10991\bin;C:\Users\Wonka\.kre\packages\KRE-CLR-x86.1.0.0-rc1-10770\bin;C:\Users\Wonka\AppData\Roaming\npm;C:\Users\Wonka\.dotnet\tools;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin;C:\Program Files\Git\bin


Initializing git credential provider...
Initialized git credential provider.


Fetching sources from git...
Cloning into 'C:\a\7ae0494'...
commit 57e32d2581b4ec58d7f0db0391ea1b3d17833503
Merge: a45f953 a714a3e
Author: Raffael Herrmann <netblognet@googlemail.com>
Date:   Wed Mar 25 14:51:49 2020 +0100

    Merge pull request #189 from csturm83/feature/single-project
    
    Feature/single project
Sources at revision 57e32d2581b4ec58d7f0db0391ea1b3d17833503.
Finished fetching sources.


Validating repository structure...
[Warning] We have detected one or more potential issues that may result in build failures.
[Warning] Deprecated project format: \QRCoderDemoUWP\project.json


Searching for projects to build...


Searching for MyGet.bat, MyGet.cmd or MyGet.ps1...


Registering NuGet package sources...
Creating new NuGet.config file in which to register package sources.
Package Source with Name: qrcoder added successfully.
Finished registering package sources.


Searching for restored NuGet packages...
Found no restored packages.


Finding and updating version manifest(s)...


Searching *.sln, *.csproj, *.vbproj or *.fsproj to patch AssemblyVersion attribute...
Found *.sln, *.csproj, *.vbproj or *.fsproj to patch AssemblyVersion attribute.
Patched AssemblyVersion attribute in file C:\a\7ae0494\QRCoderConsole\Properties\AssemblyInfo.cs
Patched AssemblyVersion attribute in file C:\a\7ae0494\QRCoderDemo\Properties\AssemblyInfo.cs
Patched AssemblyVersion attribute in file C:\a\7ae0494\QRCoderDemoUWP\Properties\AssemblyInfo.cs
Finished patching AssemblyVersion attributes.


Searching project.json files to patch...
Found project.json file(s) to patch.
[Warning] The project.json file format is deprecated. Consider upgrading to a newer Visual Studio version.
Updating version number in C:\a\7ae0494\QRCoderDemoUWP\project.json...
Updated version number in C:\a\7ae0494\QRCoderDemoUWP\project.json
Finished patching project.json files.


Searching *.vsixmanifest files to patch...
[Warning] No *.vsixmanifest files to patch.


Searching package.json files to patch...
[Warning] No package.json files to patch.


Searching composer.json files to patch...
[Warning] No composer.json files to patch.


Start running build.bat, build.cmd or build.ps1...
Running C:\a\7ae0494\MyGet.bat...

Working dir: C:\a\7ae0494
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 3/25/2020 1:52:17 PM.
     1>Project "C:\a\7ae0494\QRCoder.sln" on node 1 (Clean target(s)).
     1>ValidateSolutionConfiguration:
         Building solution configuration "Release|Any CPU".
       ValidateProjects:
         The project "QRCoderDemo" is not selected for building in solution configuration "Release|Any CPU".
         The project "QRCoderDemoUWP" is not selected for building in solution configuration "Release|Any CPU".
     1>Project "C:\a\7ae0494\QRCoder.sln" (1) is building "C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj" (3) on node 2 (Clean target(s)).
     3>CoreClean:
         Creating directory "obj\Release\".
     1>Project "C:\a\7ae0494\QRCoder.sln" (1) is building "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2) on node 1 (Clean target(s)).
     2>Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2) is building "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2:2) on node 1 (Clean target(s)).
     2>CoreClean:
         Creating directory "obj\Release\net35\".
     2>Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2) is building "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2:3) on node 1 (Clean target(s)).
     2>CoreClean:
         Creating directory "obj\Release\net452\".
   2:2>Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2:2) is building "C:\a\7ae0494\QRCoder\QRCoder.csproj" (4:4) on node 1 (Clean target(s)).
     4>CoreClean:
         Creating directory "obj\Release\net35\".
     4>Done Building Project "C:\a\7ae0494\QRCoder\QRCoder.csproj" (Clean target(s)).
     2>Done Building Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (Clean target(s)).
     2>Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2) is building "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2:4) on node 1 (Clean target(s)).
     2>CoreClean:
         Creating directory "obj\Release\netcoreapp1.1\".
     3>Project "C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj" (3) is building "C:\a\7ae0494\QRCoder\QRCoder.csproj" (4:6) on node 2 (Clean target(s)).
     4>CoreClean:
         Creating directory "obj\Release\net40\".
     4>Done Building Project "C:\a\7ae0494\QRCoder\QRCoder.csproj" (Clean target(s)).
     3>Done Building Project "C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj" (Clean target(s)).
   2:4>Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2:4) is building "C:\a\7ae0494\QRCoder\QRCoder.csproj" (4:8) on node 1 (Clean target(s)).
     4>CoreClean:
         Creating directory "obj\Release\netstandard1.1\".
     4>Done Building Project "C:\a\7ae0494\QRCoder\QRCoder.csproj" (Clean target(s)).
     2>Done Building Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (Clean target(s)).
     2>Done Building Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (Clean target(s)).
     2>Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2) is building "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2:5) on node 2 (Clean target(s)).
     2>CoreClean:
         Creating directory "obj\Release\netcoreapp2.0\".
   2:5>Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (2:5) is building "C:\a\7ae0494\QRCoder\QRCoder.csproj" (4:10) on node 2 (Clean target(s)).
     4>CoreClean:
         Creating directory "obj\Release\netstandard2.0\".
     4>Done Building Project "C:\a\7ae0494\QRCoder\QRCoder.csproj" (Clean target(s)).
     2>Done Building Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (Clean target(s)).
     2>Done Building Project "C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj" (Clean target(s)).
     1>Done Building Project "C:\a\7ae0494\QRCoder.sln" (Clean target(s)).

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.07
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj...
  Restoring packages for C:\a\7ae0494\QRCoder\QRCoder.csproj...
  Installing xunit.runner.visualstudio 2.0.1.
  Generating MSBuild file C:\a\7ae0494\QRCoder\obj\QRCoder.csproj.nuget.g.props.
  Generating MSBuild file C:\a\7ae0494\QRCoder\obj\QRCoder.csproj.nuget.g.targets.
  Restore completed in 3.46 sec for C:\a\7ae0494\QRCoder\QRCoder.csproj.
  Generating MSBuild file C:\a\7ae0494\QRCoderTests\obj\QRCoderTests.csproj.nuget.g.props.
  Generating MSBuild file C:\a\7ae0494\QRCoderTests\obj\QRCoderTests.csproj.nuget.g.targets.
  Restore completed in 3.59 sec for C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj.
PayloadGenerator.cs(1360,43): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,101): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,155): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1362,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,38): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,95): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,148): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1452,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,99): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,152): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1514,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
  QRCoder -> C:\a\7ae0494\QRCoder\bin\Release\net40\QRCoder.dll
PayloadGenerator.cs(1360,43): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,101): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,155): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1362,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,38): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,95): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,148): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1452,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,99): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,152): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1514,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
  QRCoder -> C:\a\7ae0494\QRCoder\bin\Release\net35\QRCoder.dll
PayloadGenerator.cs(1360,43): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,101): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,155): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1362,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,38): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,95): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,148): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1452,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,99): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,152): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1514,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
  QRCoder -> C:\a\7ae0494\QRCoder\bin\Release\netstandard2.0\QRCoder.dll
PayloadGeneratorTests.cs(1043,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1063,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1083,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1107,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1290,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1307,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1358,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1374,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1430,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1447,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1526,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1543,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1561,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1721,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
  QRCoderTests -> C:\a\7ae0494\QRCoderTests\bin\Release\net35\QRCoderTests.dll
PayloadGeneratorTests.cs(1043,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1063,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1083,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1107,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1290,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1307,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1358,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1374,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1430,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1447,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1526,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1543,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1561,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1721,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGenerator.cs(1360,43): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,101): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,155): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1362,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,38): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,95): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,148): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1452,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,99): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,152): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1514,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
  QRCoder -> C:\a\7ae0494\QRCoder\bin\Release\netstandard1.1\QRCoder.dll
  QRCoderTests -> C:\a\7ae0494\QRCoderTests\bin\Release\netcoreapp2.0\QRCoderTests.dll
C:\Program Files\dotnet\sdk\2.2.101\Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "NDesk.Options". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj]
Program.cs(5,7): error CS0246: The type or namespace name 'NDesk' could not be found (are you missing a using directive or an assembly reference?) [C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj]
Program.cs(231,38): error CS0246: The type or namespace name 'OptionSet' could not be found (are you missing a using directive or an assembly reference?) [C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj]
PayloadGeneratorTests.cs(1043,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1063,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1083,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1107,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1290,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1307,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1358,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1374,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1430,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1447,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1526,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1543,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1561,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1721,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
  QRCoderTests -> C:\a\7ae0494\QRCoderTests\bin\Release\net452\QRCoderTests.dll
PayloadGeneratorTests.cs(1043,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1063,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1083,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1107,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1290,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1307,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1358,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1374,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1430,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1447,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1526,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1543,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1561,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1721,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
  QRCoderTests -> C:\a\7ae0494\QRCoderTests\bin\Release\netcoreapp1.1\QRCoderTests.dll

Build FAILED.

PayloadGenerator.cs(1360,43): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,101): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,155): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1362,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,38): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,95): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,148): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1452,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,99): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,152): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1514,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,43): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,101): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,155): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1362,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,38): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,95): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,148): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1452,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,99): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,152): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1514,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,43): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,101): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,155): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1362,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,38): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,95): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,148): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1452,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,99): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,152): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1514,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGeneratorTests.cs(1043,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1063,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1083,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1107,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1290,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1307,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1358,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1374,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1430,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1447,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1526,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1543,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1561,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1721,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1043,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1063,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1083,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1107,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1290,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1307,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1358,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1374,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1430,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1447,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1526,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1543,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1561,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1721,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGenerator.cs(1360,43): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,101): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1360,155): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1362,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,38): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,95): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1388,148): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1452,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,99): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1482,152): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
PayloadGenerator.cs(1514,42): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoder\QRCoder.csproj]
C:\Program Files\dotnet\sdk\2.2.101\Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3245: Could not resolve this reference. Could not locate the assembly "NDesk.Options". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors. [C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj]
PayloadGeneratorTests.cs(1043,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1063,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1083,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1107,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1290,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1307,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1358,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1374,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1430,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1447,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1526,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1543,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1561,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1721,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1043,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1063,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1083,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singledirectdebit' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1107,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1290,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1307,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1358,61): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1374,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1430,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.periodicsinglepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1447,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1526,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1543,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1561,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
PayloadGeneratorTests.cs(1721,84): warning CS0612: 'PayloadGenerator.BezahlCode.AuthorityType.singlepayment' is obsolete [C:\a\7ae0494\QRCoderTests\QRCoderTests.csproj]
Program.cs(5,7): error CS0246: The type or namespace name 'NDesk' could not be found (are you missing a using directive or an assembly reference?) [C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj]
Program.cs(231,38): error CS0246: The type or namespace name 'OptionSet' could not be found (are you missing a using directive or an assembly reference?) [C:\a\7ae0494\QRCoderConsole\QRCoderConsole.csproj]
    105 Warning(s)
    2 Error(s)

Time Elapsed 00:00:16.06
[Error] Error during build: Build file C:\a\7ae0494\MyGet.bat exited with code 1.


Destroying git credential provider...
Destroyed git credential provider.


Cleaning up build environment...


Cleaning up sources...

Build steps: 19
  step 0 took 00:00:00.2298739: Package Source Verification
  step 1 took 00:00:00.0009038: Build environment configuration
  step 2 took 00:00:00.0000936: Git configuration
  step 3 took 00:00:00.0000027: Set user-defined environment variables
  step 4 took 00:00:05.0512220: Git credential provider initialization
  step 5 took 00:00:04.1569743: Git repository clone
  step 6 took 00:00:00.0029133: Repository Verification
  step 7 took 00:00:00.0061992: Project lookup
  step 8 took 00:00:02.2298939: NuGet configuration
  step 9 took 00:00:00.0000019: Package restore
  step 10 took 00:00:00.0012787: Find restored packages
  step 11 took 00:00:00.0370518: Manifest patching
  step 12 took 00:00:00.0000022: Project compilation
  step 13 took 00:00:00.0000007: Test execution
  step 14 took 00:00:17.8654326: Custom build script execution
  step 14 took 00:00:00.0000015: Git repository labeling
  step 15 took 00:00:04.0447785: Git credential provider cleanup
  step 16 took 00:00:00.0005355: Environment cleanup
  step 17 took 00:00:00.0000423: Source cleanup
[Error] BUILD FAILED
```

At the beginning of the log I see, that "QRCoderDemo" and "QRCoderDemoUWP" won't be builded per definition. That's good, because the MyGet builds shall be used only as "is it compileable" check for the main library and as source for the Nuget package. The Nuget package should not contain the demo projects, but only the QRCoder libs. 

```Powershell
       ValidateProjects:
         The project "QRCoderDemo" is not selected for building in solution configuration "Release|Any CPU".
         The project "QRCoderDemoUWP" is not selected for building in solution configuration "Release|Any CPU".
```

So I'll try to change the project definition so that the Console project also gets recognized as "no selected for building". Right?

codebude commented:
Ok, removing the QRCoderConsole project from the automatic MyGet build (by removing it from the build chain as shown here: https://github.com/codebude/QRCoder/commit/17c8fd356dde5567158a13d56ac8878466815098 ) did the trick. The project can be built on MyGet again.

jnyrup commented:
`dotnet build` won't restore nuget packages for old style csproj such as `QRCoderConsole`.
So `NDesk.options` is not restored and the build of `QRCoderConsole` fails.
*If* you wanted to do that, you can run `nuget restore QRCoder.sln` before `dotnet build`.

codebude commented:
Thanks for clarification - good to know why things failed. That would solve the problem described above, too. But since the flow looks like this...

Git commit --> webhook --> MyGet pulls code from Github --> MyGet builds QRCoder library --> (manual step) I decide to push one of the MyGet builds to Nuget

...a build of the console project isn't needed at all on MyGet. :-)
